File: tournaments/kthDigit/kthDigit.cs
int kthDigit(int n, int k) {
    var numDigits = 0;
    var number = n;

    while (0 < number) {
        ++numDigits;
        number /= 10;
    }
    
    int indexFromLast = numDigits - k + 1;
    
    while (0 < n) {
        if (--indexFromLast == 0) {
            return n % 10;
        }

        n /= 10;
    }
    
    return -1;
}

File: tournaments/kthDigit/kthDigit.js
function kthDigit(n, k) {
    const b = n.toString().length;
    let a = b - k + 1;

    while (n) {
        if (!--a)
            return n % 10;
        n = ~~(n / 10);
    }

    return -1;
}
